src/Domain/class-wc-email-customer-new-account-blocks.php

```diff
@@ -0,0 +1,190 @@
+		 */
+		public $default_template_path;
+
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+			// Note - we're using the same ID as the real email.
+			// This ensures that any merchant tweaks (Settings > Emails)
+			// apply to this email (consistent with the core email).
+			$this->id                    = 'customer_new_account';
+			$this->customer_email        = true;
+			$this->title                 = __( 'New account', 'woo-gutenberg-products-block' );
+			$this->description           = __( 'Customer "new account" emails are sent to the customer when a customer signs up via checkout or account blocks.', 'woo-gutenberg-products-block' );
+			$this->template_html         = 'emails/customer-new-account-blocks.php';
+			$this->template_plain        = 'emails/plain/customer-new-account-blocks.php';
+			$this->default_template_path = plugin_dir_path( __FILE__ ) . 'templates/';
+
+			// Call parent constructor.
+			parent::__construct();
+		}
+
+		/**
+		 * Get email subject.
+		 *
+		 * @since  3.1.0
+		 * @return string
+		 */
+		public function get_default_subject() {
+			return __( 'Your {site_title} account has been created!', 'woo-gutenberg-products-block' );
+		}
+
+		/**
+		 * Get email heading.
+		 *
+		 * @since  3.1.0
+		 * @return string
+		 */
+		public function get_default_heading() {
+			return __( 'Welcome to {site_title}', 'woo-gutenberg-products-block' );
+		}
+
+		/**
+		 * Trigger.
+		 *
+		 * @param int    $user_id User ID.
+		 * @param string $user_pass User password.
+		 * @param bool   $password_generated Whether the password was generated automatically or not.
+		 */
+		public function trigger( $user_id, $user_pass = '', $password_generated = false ) {
+			$this->setup_locale();
+
+			if ( $user_id ) {
+				$this->object = new WP_User( $user_id );
+
+				// Generate a magic link so user can set initial password.
+				$key = get_password_reset_key( $this->object );
+				if ( ! is_wp_error( $key ) ) {
+					$this->set_password_url = network_site_url(
+						"wp-login.php?action=rp&key=$key&login=" . rawurlencode( $this->object->user_login ),
+						'login'
+					);
+				}
+
+				$this->user_login = stripslashes( $this->object->user_login );
+				$this->user_email = stripslashes( $this->object->user_email );
+				$this->recipient  = $this->user_email;
+			}
+
+			if ( $this->is_enabled() && $this->get_recipient() ) {
+				$this->send( $this->get_recipient(), $this->get_subject(), $this->get_content(), $this->get_headers(), $this->get_attachments(), $this->set_password_url );
+			}
+
+			$this->restore_locale();
+		}
+
+		/**
+		 * Get content html.
+		 *
+		 * @return string
+		 */
+		public function get_content_html() {
+			return wc_get_template_html(
+				$this->template_html,
+				array(
+					'email_heading'      => $this->get_heading(),
+					'additional_content' => $this->get_additional_content(),
+					'user_login'         => $this->user_login,
+					'user_pass'          => $this->user_pass,
```

Comment:
Should we exclude `user_pass` from template args here because that's not provided (same applies to `get_content_plain`)?

Reply:
Good spotting! 4219194f8294b9fe1b10d25728b7a88962bbcbdc